boards: Add esp32c3_lckfb board support and QMI8658C sensor driver

godmial opened the pull request:
## Summary

This PR adds support for the LCKFB ESP32-C3 development board (esp32c3_lckfb) and implements a sensor driver for the QMI8658C 6-axis IMU sensor.

## Changes

### Board Support
- Added complete BSP support for esp32c3_lckfb board
  - Board configuration files (Kconfig, board.cmake, board.yml)
  - Device tree files (DTS, pinctrl)
  - Board documentation
  - OpenOCD debugging configuration
- Fixed BSP configuration to use generic ESP32-C3 SOC selection
- Updated vendor identifier to follow Zephyr convention (espressif)

### Sensor Driver
- Implemented QMI8658C 6-axis IMU sensor driver
  - Supports 3-axis accelerometer (±4g range)
  - Supports 3-axis gyroscope (±512dps range)
  - I2C interface support (address 0x6A)
  - Device tree bindings
- Added sample application demonstrating sensor usage
  - Polling mode operation at 10 Hz
  - Displays acceleration, angular velocity, and calculated tilt angles
  - Includes board-specific overlay for esp32c3_lckfb

## Testing

- Board BSP tested with QMI8658C sensor sample
- Sensor driver verified with sample application on esp32c3_lckfb board
- Build and flash operations successful

## References

- LCKFB ESP32-C3 Development Board: https://wiki.lckfb.com/zh-hans/szpi-esp32c3/

godmial committed:
新增qmi8658c，qmc5883l驱动，以及立创实战派C3的bsp

godmial committed:
boards: esp32c3_lckfb: fix BSP configuration and sample overlay

This commit fixes several configuration issues in the esp32c3_lckfb BSP:

1. Kconfig: Changed SOC selection from SOC_ESP32C3_WROOM_02_N8 to
   SOC_ESP32C3 to use generic ESP32-C3 configuration without specifying
   a particular module type.

2. Device Tree: Updated DTS include from esp32c3_wroom_n4.dtsi to
   esp32c3_common.dtsi to match the generic SOC selection in Kconfig.

3. Vendor: Changed vendor from 'lckfb' to 'espressif' in both board.yml
   and esp32c3_lckfb.yaml to follow Zephyr convention where all ESP32
   boards use 'espressif' as the vendor identifier.

4. Sample Overlay: Added device tree alias and node label in
   qmi8658c sample overlay to fix compilation error where code uses
   DT_ALIAS(qmi8658c_0) but the alias was not defined.

These changes ensure consistency with other ESP32-C3 boards in Zephyr
and fix build issues with the QMI8658C sensor sample.

Signed-off-by: Your Name <your.email@example.com>
Signed-off-by: godmial <wuyingxiang58@gmail.com>

JarmouniA commented on boards/espressif/esp32c3_lckfb/board.yml:
The `boards/espressif` folder is only supposed to contain official Espressif development boards, which is not the case for this board.

godmial replied:
Thank you both for your feedback! @JarmouniA @marekmatej

I appreciate your guidance on the proper directory structure. After careful consideration, I'd like to create a dedicated `boards/lckfb` folder for the following reasons:

LCKFB (https://lckfb.com/) is an open platform that provides a wide range of development boards across multiple chip families, including:
- GD series (GigaDevice)
- STM32 series (STMicroelectronics)
- ESP32 series (Espressif)
- TI series (Texas Instruments)
- And more

Given the platform's commitment to supporting diverse hardware and the potential for multiple board variants, I believe a dedicated vendor directory would be more appropriate than placing boards under `boards/others`.

I'm committed to actively contributing board support for additional LCKFB boards in the future. I also plan to:
1. Document the lessons learned from this PR process
2. Share this experience with other developers in the LCKFB community who are interested in contributing
3. Actively participate in future board adaptation efforts

This approach will help establish a clear structure for LCKFB boards in Zephyr and make it easier for other contributors to add support for additional boards from this platform.

I'll proceed with creating the `boards/lckfb` directory structure. Thank you again for your valuable input!

godmial committed:
boards: lckfb: move esp32c3_lckfb BSP and fix compliance issues

This commit moves the esp32c3_lckfb BSP from boards/espressif/ to
boards/lckfb/ as requested by Zephyr project maintainers, and fixes
all compliance issues identified by CI checks.

Changes:

1. BSP relocation:
   - Move BSP from boards/espressif/esp32c3_lckfb to boards/lckfb/esp32c3_lckfb
   - Update vendor from 'espressif' to 'lckfb' in board.yml and esp32c3_lckfb.yaml

2. Compliance fixes:
   - Remove trailing blank lines from all files (20+ files)
   - Remove trailing whitespace from DTS and C source files
   - Fix sorting in drivers/sensor/CMakeLists.txt and Kconfig
   - Fix YAML formatting issues (remove extra blank lines)
   - Add SPDX license header to dts/bindings/vendor-prefixes.txt

3. Documentation:
   - Restore doc/build/ directory from main branch

These changes ensure the BSP follows Zephyr conventions and passes all
compliance checks.

godmial committed:
boards: lckfb: move esp32c3_lckfb BSP and fix compliance issues

This commit moves the esp32c3_lckfb BSP from boards/espressif/ to
boards/lckfb/ as requested by Zephyr project maintainers, and fixes
all compliance issues identified by CI checks.

Changes:

1. BSP relocation:
   - Move BSP from boards/espressif/esp32c3_lckfb to boards/lckfb/esp32c3_lckfb
   - Update vendor from 'espressif' to 'lckfb' in board.yml and esp32c3_lckfb.yaml

2. Compliance fixes:
   - Remove trailing blank lines from all files (20+ files)
   - Remove trailing whitespace from DTS and C source files
   - Fix sorting in drivers/sensor/CMakeLists.txt and Kconfig
   - Fix YAML formatting issues (remove extra blank lines)
   - Add SPDX license header to dts/bindings/vendor-prefixes.txt

3. Documentation:
   - Restore doc/build/ directory from main branch

These changes ensure the BSP follows Zephyr conventions and passes all
compliance checks.

sylvioalves commented on boards/lckfb/esp32c3_lckfb/esp32c3_lckfb-pinctrl.dtsi:
```diff
@@ -0,0 +1,60 @@
+		};
+	};
+
+	twai_default: twai_default {
```
I guess you should remove this as those pins GPIO4/GPIO5 are attached to SPI/LCD.

sylvioalves commented on boards/lckfb/esp32c3_lckfb/esp32c3_lckfb.dts:
```diff
@@ -0,0 +1,112 @@
+		i2c-0 = &i2c0;
+		watchdog0 = &wdt0;
+	};
+
```
What about LCD DC and BACKLIGHT pins?
